docs/events/glides/com.mbta.ctd.glides.vehicle_trip_assignments.v1.mdx

```diff
@@ -0,0 +1,176 @@
+- (In both cases, the `serviceDate` must also match in order to know the two TripKeys refer to the same trip.)
+
+The difference with the TripKey in `trips_updated` is because `trips_updated` has to give enough information about a trip that a consumer can understand it even without recognizing the trip ID, while `vehicle_trip_assignments` assumes that you can look up information in `trips_updated` or in `GTFS`, so doesn't give as much information as possible about the trip. (Though consumers should keep in mind that order of events between two event streams is not guaranteed, and a trip that's referenced in this feed may not have appeared in `trips_updated`.)
+
+For scheduled trips, the `tripId` is intended to line up to the trip IDs in GTFS most of the time, because they're derived from the same source (HASTUS's `trp_int_number`). However, Glides may be using a slightly different schedule than any consumer (due to disruptions, for example), so matching GTFS is not guaranteed. If a consumer sees a `tripId` in this feed that also appears in GTFS on a trip on the same service date, then it refers to the same trip.
+
+For added trips, Glides makes up a new ID, which is kept consistent for any subsequent updates to the same added trip. Glides tends to prefix trip IDs for added trips with the string `"ADDED-"` but consumers MUST NOT assume any particular format, or assume that they can tell an ID for an added trip apart from a scheduled trip ID by the string alone. That's what the `scheduled` field is for.
+
+The added/scheduled distinction is according to the schedule in Glides. If Glides has a different schedule than RTR or GTFS, then a trip that Glides thinks is scheduled may not match up with scheduled trip in GTFS, and RTR (or other consumers) may decide that a trip that was added in Glides lines up close enough to correspond to a scheduled trip in GTFS.
+
+Revenue is included because this is the most important information that Glides can provide to riders about a train, and its presence will make it less likely for a consumer of this feed to incorrectly interpret a nonrevenue train as being in service. Especially since nonrevenue trips are not in GTFS and their IDs are likely to not be recognized by a consumer.
+
+Consumers SHOULD use the entire object when comparing two `TripKey`s to see if they're equal, not just the `tripId` field. Otherwise, updates could be applied to the wrong trip, for example, updating the vehicle assignment for yesterday's trip instead of today's.
+
+The `revenue` and `scheduled` fields use string enums instead of booleans so that other states could be added in the future. Clients MUST tolerate values for these strings that they don't recognize. A safe way to handle this would be to treat the TripKey like an unrecognized trip, or to treat it like it's `null` i.e. the trip doesn't exist and a vehicle assigned to it is treated like it's unassigned from any trips.
```

Comment:
Do you have thoughts on what other states we might want in these fields at a later date? It's easy enough to be tolerant, but "Is this on the schedule" or "is this revenue generating" feel like binary questions to me, and RTR treats them as such.

Reply:
For "scheduled", I could imagine a "added, but inferred by Glides as opposed to entered by an inspector" state, or a "in Glides' HASTUS schedule but we already checked and know it won't be in GTFS" state.

For "revenue", I can't think of any third states, but this matches the enum used in [`trips_updated`](https://mbta.github.io/schemas/events/glides/com.mbta.ctd.glides.trips_updated.v1#scheduled-trip).

In both cases, it's mostly an attempt to support future states we _haven't_ thought of yet. They're definitely binary now, and Glides also treats them as booleans, but if the situation arises, it's easy to update applications to change the type to an enum, and much harder to change the events. Doing it this way to start with saves us a couple dance moves in a hypothetical future migration.

Reply:
revenue was removed in 22245e1. Now it's just the `scheduled` field that's the enum.

Also "inferred" wouldn't be a good example of a new state because that should be part of the data, not part of the trip ID. But "automatically created placeholder" and "scheduled but not in GTFS" might still be flavors of ID.

I'd still like to keep the enum, but less strongly than when we also had `revenue`.